include/fc/static_variant.hpp

```diff
@@ -181,7 +181,7 @@ class static_variant {
     static_assert(impl::type_info<Types...>::no_reference_types, "Reference types are not permitted in static_variant.");
     static_assert(impl::type_info<Types...>::no_duplicates, "static_variant type arguments contain duplicate types.");
 
-    int _tag;
+    size_t _tag;
```

Comment:
How does `size_t` get serialized? It's not safe to change the type from signed to unsigned when it's already in production, since it may break consensus. Safer to change to `int64_t` as already done in Steem: https://github.com/steemit/fc/commit/b4fd91d889796c6aee84428427552046e5316392. Related discussion in https://github.com/steemit/fc/pull/37.

Reply:
Pretty sure it's as safe as it could get. The tag in static_variant gets assigned at compile time, it is protected with compile time asserts. The tag should not go beyond the [0, N) boundary and if it goes  somehow (presumably due to broken deserialization or some intentional abuse) you'll get a runtime exception. 

But perhaps you have better perspective and we'll surely do as you suggest.

Reply:
Yes, maybe there are intentional abuses, e.g. comparison, `0<-1` is `false` for `int64_t` but `true` for `uint64_t` (perhaps runtime error for `size_t`?)